code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -571,17 +577,21 @@
 	if(!is_species(H, /datum/species/jelly/luminescent) || !species)
 		return
 	CHECK_DNA_AND_SPECIES(H)
-
+	
 	if(species.current_extract)
 		species.extract_cooldown = world.time + 100
+
+	if(species.current_extract && H.blood_volume >= blood_cost)
 		var/cooldown = species.current_extract.activate(H, species, activation_type)
```

Comment:
```suggestion
	if(species.current_extract && H.blood_volume >= blood_cost)
		species.extract_cooldown = world.time + 100
		var/cooldown = species.current_extract.activate(H, species, activation_type)
```
otherwise, you'll get put on a 10 second cooldown if you try to use an extract while you don't have enough blood to pay its blood cost.